libraries/src/Form/FormField.php

```diff
@@ -1380,7 +1397,7 @@ protected function getLayoutData()
 			'validationtext' => $this->validationtext,
 			'readonly'       => $this->readonly,
 			'repeat'         => $this->repeat,
-			'required'       => (bool) $this->required,
+			'required'       => (bool) $this->required ||(bool) $this->requireon,
```

Comment:
```suggestion
			'required'       => (bool) $this->required || (bool) $this->requireon,
```
Just a small code style improvement.